db/config.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package db
+
+import (
+	"context"
+
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/acp"
+	"github.com/sourcenetwork/defradb/events"
+	"github.com/sourcenetwork/defradb/lens"
+)
+
+const (
+	defaultMaxTxnRetries  = 5
+	updateEventBufferSize = 100
+)
+
+// Option is a funtion that sets a config value on the db.
+type Option func(*db)
+
+// WithACP enables access control. If path is empty then acp runs in-memory.
+func WithACP(path string) Option {
```

Comment:
thought: Seeing these 'new' ACP funcs was a surprise to me when reviewing, and there is not really any documentation in the PR description, and it is done in the same commit as the new Lens stuff.  This forces reviewers to spend more time/energy figuring out what is existing stuff being moved, and what is new stuff being added.  I suggest (in the future, not asking to to change this now) separate commits for this kind of stuff, and please be a little more explicit in the PR description as to what has been done.